src/coreclr/src/tools/aot/crossgen2/Program.cs

```diff
@@ -347,6 +348,27 @@ private int Run()
                         }
                     }
 
+                    foreach (string inputFilePath in allInputFilePaths.Values)
+                    {
+                        EcmaModule module = typeSystemContext.GetModuleFromPath(inputFilePath);
+                        foreach (MetadataType type in module.GetAllTypes())
```

Comment:
I did not meant to scan all methods and in all assemblies. That is very expensive. I meant to just check whether the assembly is managed C++ (you can do that by looking at PE header).

Reply:
So far I have only found the bit <code>CorFlags.NativeEntryPoint</code>. Is that a reliable Managed C++ indicator? Other than that I don't see any PE Characteristics, DllCharacteristics bit flags or assembly-level attributes usable for this detection. Are you aware of any existing code that performs this detection so that I can take a look and adapt it for use in CG2?

Reply:
When both `COMIMAGE_FLAGS_ILONLY` and `COMIMAGE_FLAGS_IL_LIBRARY` bits in CorHeader are zero, it means the binary is MC++.

Reply:
Thanks Jan, fixed in 3rd commit.